`cargo test --no-default-features --features=full` fails because of `no_std`

When avoiding `std`, and asking for `--features=full`, some tests fail:

```
dkg@bob:~/src/derive_more$ cargo test --no-default-features --features=full
   Compiling derive_more v1.0.0 (/home/dkg/src/derive_more)
error: cannot find macro `vec` in this scope
   --> tests/lib.rs:270:19
    |
270 |     let int_vec = vec![MyInt2(2), MyInt2(3)];
    |                   ^^^

error: cannot find macro `vec` in this scope
   --> tests/lib.rs:261:28
    |
261 |     let mut my_vec = MyVec(vec![5, 8]);
    |                            ^^^

error: cannot find macro `format` in this scope
   --> tests/lib.rs:200:23
    |
200 |     assert_eq!("100", format!("{:b}", MyInt(4)));
    |                       ^^^^^^

error: cannot find macro `format` in this scope
   --> tests/lib.rs:199:22
    |
199 |     assert_eq!("36", format!("{:o}", MyInt(30)));
    |                      ^^^^^^

error[E0412]: cannot find type `Vec` in this scope
  --> tests/lib.rs:43:14
   |
43 | struct MyVec(Vec<i32>);
   |              ^^^ not found in this scope

error[E0412]: cannot find type `Box` in this scope
  --> tests/lib.rs:49:19
   |
49 | struct MyBoxedInt(Box<i32>);
   |                   ^^^ not found in this scope

error[E0599]: no method named `to_string` found for struct `MyInt` in the current scope
   --> tests/lib.rs:198:32
    |
24  | struct MyInt(i32);
    | ------------ method `to_string` not found for this struct
...
198 |     assert_eq!("30", MyInt(30).to_string());
    |                                ^^^^^^^^^ method not found in `MyInt`
   --> /usr/src/rustc-1.81.0/library/alloc/src/string.rs:2544:8
    |
    = note: the method is available for `MyInt` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `ToString` which provides `to_string` is implemented but not in scope; perhaps you want to import it
    |
3   + use serde::__private::ToString;
    |

error[E0599]: no method named `to_string` found for struct `SimpleStruct` in the current scope
   --> tests/lib.rs:207:46
    |
71  | struct SimpleStruct {
    | ------------------- method `to_string` not found for this struct
...
207 |     assert_eq!("5", SimpleStruct { int1: 5 }.to_string());
    |                                              ^^^^^^^^^ method not found in `SimpleStruct`
   --> /usr/src/rustc-1.81.0/library/alloc/src/string.rs:2544:8
    |
    = note: the method is available for `SimpleStruct` here
    |
    = help: items from traits can only be used if the trait is in scope
help: trait `ToString` which provides `to_string` is implemented but not in scope; perhaps you want to import it
    |
3   + use serde::__private::ToString;
    |

error[E0599]: no method named `into_iter` found for reference `&MyVec` in the current scope
   --> tests/lib.rs:266:26
    |
266 |     assert_eq!((&my_vec).into_iter().next(), Some(&20));
    |                          ^^^^^^^^^ method not found in `&MyVec`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `into_iter`, perhaps you need to implement it:
            candidate #1: `IntoIterator`

error[E0599]: no method named `into_iter` found for mutable reference `&mut MyVec` in the current scope
   --> tests/lib.rs:267:30
    |
267 |     assert_eq!((&mut my_vec).into_iter().next(), Some(&mut 20));
    |                              ^^^^^^^^^ method not found in `&mut MyVec`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `into_iter`, perhaps you need to implement it:
            candidate #1: `IntoIterator`

error[E0599]: no method named `into_iter` found for struct `MyVec` in the current scope
   --> tests/lib.rs:268:23
    |
43  | struct MyVec(Vec<i32>);
    | ------------ method `into_iter` not found for this struct
...
268 |     assert_eq!(my_vec.into_iter().next(), Some(20));
    |                       ^^^^^^^^^ method not found in `MyVec`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `into_iter`, perhaps you need to implement it:
            candidate #1: `IntoIterator`

error[E0433]: failed to resolve: use of undeclared type `Box`
   --> tests/lib.rs:277:32
    |
277 |     let mut boxed = MyBoxedInt(Box::new(5));
    |                                ^^^ use of undeclared type `Box`

error[E0433]: failed to resolve: use of undeclared type `Box`
   --> tests/lib.rs:280:27
    |
280 |     assert_eq!(MyBoxedInt(Box::new(7)), boxed)
    |                           ^^^ use of undeclared type `Box`

Some errors have detailed explanations: E0412, E0433, E0599.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `derive_more` (test "lib") due to 13 previous errors
warning: build failed, waiting for other jobs to finish...
101 dkg@bob:~/src/derive_more$ 
```

I think either the tests that fail should be marked as only applicable when `std` is present, or `std` should be added to the `full` featureset.  There are probably other ways to resolve the issue too, that someone with deeper rust/cargo knowledge than me could come up with.

--- a/ci/test_all_features.sh
+++ b/ci/test_all_features.sh
@@ -7,6 +7,6 @@ fi
 
 set -euxo pipefail
 
-for feature in $(tomljson Cargo.toml | jq --raw-output '.features | keys[]' | grep -v 'default\|std\|full\|testing-helpers'); do
+for feature in $(tomljson Cargo.toml | jq --raw-output '.features | keys[]' | grep -v 'default\|std\|testing-helpers'); do
     RUSTFLAGS='-D warnings' cargo +nightly test -p derive_more --tests --no-default-features --features "$feature$std,testing-helpers"
 done

--- a/tests/lib.rs
+++ b/tests/lib.rs
@@ -1,5 +1,12 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
+#[cfg(not(feature = "std"))]
+#[macro_use]
+extern crate alloc;
+
+#[cfg(not(feature = "std"))]
+use alloc::{boxed::Box, string::ToString as _, vec::Vec};
+
 use derive_more::{
     Add, AddAssign, Binary, BitAnd, BitOr, BitXor, Constructor, Deref, DerefMut,
     Display, Div, From, FromStr, Index, IndexMut, Into, IntoIterator, Mul, MulAssign,

